RecoMuon/MuonRechitClusterProducer/plugins/RechitClusterProducer.cc

```diff
@@ -0,0 +1,268 @@
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/MuonReco/interface/MuonRecHitCluster.h"
+#include "Geometry/CSCGeometry/interface/CSCGeometry.h"
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+#include "DataFormats/MuonDetId/interface/CSCDetId.h"
+
+#include "fastjet/JetDefinition.hh"
+#include "fastjet/ClusterSequence.hh"
+#include "fastjet/PseudoJet.hh"
+
+#include "Geometry/CSCGeometry/interface/CSCGeometry.h"
+#include "Geometry/DTGeometry/interface/DTGeometry.h"
+#include "DataFormats/MuonDetId/interface/CSCDetId.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Math/interface/PtEtaPhiMass.h"
+#include "DataFormats/DTRecHit/interface/DTRecHitCollection.h"
+#include "DataFormats/CSCRecHit/interface/CSCRecHit2DCollection.h"
+#include "DataFormats/MuonReco/interface/MuonRecHitCluster.h"
+
+typedef std::vector<reco::MuonRecHitCluster> RecHitClusterCollection;
+
+template <typename Trait>
+class RechitClusterProducerT : public edm::global::EDProducer<> {
+  typedef typename Trait::RecHitRef RechitRef;
+  typedef typename Trait::RecHitRefVector RecHitRefVector;
+
+public:
+  explicit RechitClusterProducerT(const edm::ParameterSet&);
+  ~RechitClusterProducerT() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+  const edm::ESGetToken<typename Trait::GeometryType, MuonGeometryRecord> GeometryToken_;
+  edm::EDGetTokenT<typename Trait::InputType> InputToken_;
+  typedef std::vector<reco::MuonRecHitCluster> RecHitClusterCollection;
+
+  const double rParam_;      // distance paramter
+  const int nRechitMin_;     // min number of rechits
+  const int nStationThres_;  // min number of rechits to count towards nStation
+};
+
+template <typename Trait>
+RechitClusterProducerT<Trait>::RechitClusterProducerT(const edm::ParameterSet& iConfig)
+    : GeometryToken_(esConsumes<typename Trait::GeometryType, MuonGeometryRecord>()),
+      InputToken_(consumes<typename Trait::InputType>(iConfig.getParameter<edm::InputTag>("recHitLabel"))),
+      rParam_(iConfig.getParameter<double>("rParam")),
+      nRechitMin_(iConfig.getParameter<int>("nRechitMin")),
+      nStationThres_(iConfig.getParameter<int>("nStationThres")) {
+  produces<RecHitClusterCollection>();
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+template <typename Trait>
+void RechitClusterProducerT<Trait>::produce(edm::StreamID, edm::Event& ev, const edm::EventSetup& iSetup) const {
+  auto const& geo = iSetup.getData(GeometryToken_);
+  auto const& rechits = ev.get(InputToken_);
+
+  std::vector<fastjet::PseudoJet> fjInput;
+  RecHitRefVector inputs_;
+
+  fastjet::JetDefinition jet_def(fastjet::cambridge_algorithm, rParam_);
+
+  inputs_.clear();
+  fjInput.clear();
+
+  Trait RecHitTrait;
```

Comment:
```suggestion
  Trait recHitTrait;
```
start local variables and data members with lower case letter. Capitalization is reserved for class names.